scalyr_agent/builtin_monitors/openmetrics_monitor.py

```diff
@@ -0,0 +1,599 @@
+        base_request_headers.update(self.__headers)
+
+        self.__session = requests.Session()
+        self.__session.headers.update(base_request_headers)
+
+    def gather_sample(self):
+        # type: () -> None
+        metrics = self._scrape_metrics(self.__url)
+
+        for metric_name, extra_fields, metric_value in metrics:
+            extra_fields.update(self.__base_extra_fields or {})
+            self._logger.emit_value(
+                metric_name, metric_value, extra_fields=extra_fields
+            )
+
+    def check_connectivity(self) -> requests.Response:
+        """
+        Send the request to the configured URL and return a response.
+
+        This is used for checking connectivity and other similar purposes.
+        """
+        request_kwargs = self._get_request_kwargs(url=self.__url)
+        resp = self.__session.get(self.__url, **request_kwargs, timeout=self.__timeout)
+        return resp
+
+    def _get_request_kwargs(self, url: str) -> dict:
+        """
+        Return optional keyword arguments which are passed to the requests.get() method for the
+        provided url.
+        """
+        request_kwargs = {}
+        if url.startswith("https://"):
+            verify = self.__verify_https
+
+            if self.__ca_file:
+                verify = self.__ca_file
+
+            request_kwargs["verify"] = verify
+
+        # TODO: Use rate limited warn if verify is false
+        return request_kwargs
+
+    def _scrape_metrics(self, url):
+        # type: (str) -> List[Tuple[str, dict, Any]]
+        """
+        Scrape metrics from Prometheus interface and return dictionary with parsed metrics.
+        """
+        request_kwargs = self._get_request_kwargs(url=url)
+
+        try:
+            resp = self.__session.get(url, **request_kwargs, timeout=self.__timeout)
+        except Exception as e:
+            self._logger.warn(
+                "Outgoing request failed to %s failed: %s" % (url, str(e))
+            )
+            return []
+
+        if resp.status_code != 200:
+            self._logger.warn(
+                "Failed to fetch metrics from %s: status_code=%s,body=%s"
+                % (self.__url, resp.status_code, resp.text)
+            )
+            return []
+
+        # NOTE: To be more compatible we assume Content-Type is text in case header is not provided
+        # and we only explicitly skip Protobuf format
+        content_type = resp.headers.get("Content-Type", "application/openmetrics-text")
+        if "application/openmetrics-protobuf" in content_type:
+            self._logger.warn(
+                "Received non-supported protobuf format. Content-type: %s"
+                % (content_type)
+            )
+            return []
+
+        lines = resp.text.splitlines()
+
+        result = []
+
+        metric_name_to_type_map = {}
+
+        metrics_count = 0
+
+        for line in lines:
+            comment_metric_name = None
```

Comment:
it seems like those 2 variables are redeclared inside of the if statement below, so the current ones can be removed.

Reply:
Good catch, will remove it.